### PR TITLE
[full-ci] Update drone starlark

### PR DESCRIPTION
**Pull request by amrita-shrestha**
## Description
This is an update on drones.star file 
- using drone-cancel-previous-builds
- using thegeeklab/drone-github-comment instead of jmccann/drone-github-comment for github comments 
- moving docker image references to constants
- using caches images

## Related Issue
- Part of issue: 
- https://github.com/owncloud/QA/issues/700
- https://github.com/owncloud/QA/issues/717
- https://github.com/owncloud/QA/issues/729
- https://github.com/owncloud/QA/issues/728

### CODE REVIEW

**Comment by SwikritiT** [comment]
I think this file can be removed too https://github.com/owncloud/web/blob/master/tests/acceptance/cancelBuilds.sh

**Review by kiranparajuli589 — COMMENTED** [review]
lgtm :+1:

**Comment by kiranparajuli589** [comment]
GitHub comment step was not happy here https://drone.owncloud.com/owncloud/web/23474/68/20

**Comment by ownclouders** [comment]
:boom: Acceptance test [<strong>e2e-tests OC10</strong>](https://drone.owncloud.com/owncloud/web/23552/10/1) failed. Further test are cancelled...

File: .drone.star [hunk 1]
```diff
@@ -11,6 +11,30 @@ OC_CI_PHP = "owncloudci/php:7.4"
...
+OC_CI_HUGO = "owncloudci/hugo:0.89.4"
+PLUGINS_GH_PAGES = "plugins/gh-pages:1"
+PLUGINS_DOWNSTREAM = "plugins/downstream"
+MINIO_MC_RELEASE_2020_VERSION = "minio/mc:RELEASE.2020-12-10T01-26-17Z"
+PLUGINS_S3 = "plugins/s3"
+THEGEEKLAB_DRONE_GITHUB_COMMENT = "thegeeklab/drone-github-comment:1"
+ALPINE_GIT = "alpine/git:latest"
+OC_CI_DRONE_ANSIBLE = "owncloudci/drone-ansible:latest"
+OC_CI_BAZEL_BUILDIFIER = "owncloudci/bazel-buildifier"
+OC_CI_DRONE_SKIP_PIPLELINE = "owncloudci/drone-skip-pipeline"
```
Comment (SwikritiT) [inline .drone.star:35]:
```suggestion
OC_CI_DRONE_SKIP_PIPELINE = "owncloudci/drone-skip-pipeline"
```

File: .drone.star [hunk 2]
```diff
@@ -2789,7 +2806,6 @@ def githubComment(alternateSuiteName):
         ],
         "when": {
             "status": [
-                "success",
```
Comment (pascalwengerter) [inline .drone.star:2792]: Why are we dropping the `success` message here?

File: .drone.star [hunk 3]
```diff
@@ -838,29 +862,22 @@ def yarnlint(ctx):
...
     return [{
         "kind": "pipeline",
         "type": "docker",
-        "name": "stop-recent-builds",
-        "workspace": {
-            "base": dir["base"],
-            "path": config["app"],
+        "name": "cancel-previous-builds",
+        "clone": {
+            "disable": True,
```
Comment (pascalwengerter) [inline .drone.star:871]: What are we introducing here?

**Review by pascalwengerter — COMMENTED** [review]
Looks good so far, only small questions from my side:)

File: .drone.star (hunk 2 above)
Comment (amrita-shrestha) [reply .drone.star:2792]: we only want to comment when the test fails. Other repos have a similar approach.

File: .drone.star (hunk 3 above)
Comment (kiranparajuli589) [reply .drone.star:871]: To check and cancel the previous running build, the pipeline doesn't need to clone the actual code

**Comment by kiranparajuli589** [comment]
failed at https://drone.owncloud.com/owncloud/web/23656/10/17
reported here https://github.com/owncloud/web/issues/6596
build restarted

**Comment by kiranparajuli589** [comment]
again failed at the same place, should we skip it for now? @individual-it @amrita-shrestha @saw-jan @SwikritiT

File: .drone.star [hunk 4]
```diff
@@ -3,14 +3,38 @@ FULL = 1
 FEDERATED = 2
 NOTIFICATIONS = 3
 
+ALPINE_GIT = "alpine/git:latest"
+DEEPDRIVER_DOCKER_ORACLE_XE_11G = "deepdiver/docker-oracle-xe-11g:latest"
+DRONE_CLI_ALPINE = "drone/cli:alpine"
+HENRICH_DOCKER_SAUCE_CONNECT = "henrrich/docker-sauce-connect:latest"
+MELTWATER_DRONE_CACHE = "meltwater/drone-cache:v1"
+MINIO_MC_RELEASE_2020_VERSION = "minio/mc:RELEASE.2020-12-10T01-26-17Z"
+MINIO_MC_RELEASE_2021_VERSION = "minio/mc:RELEASE.2021-03-23T05-46-11Z"
```
Comment (individual-it) [inline .drone.star:12]: do we need both versions or can we use the same in all cases?

**Comment by individual-it** [comment]
I would suggest to find the reason for it

**Comment by amrita-shrestha** [comment]
According to fschade, e2e intermittent test will be solved by https://github.com/owncloud/web/pull/6603 as mentioned in
[[fschade](https://github.com/fschade)](https://github.com/owncloud/web/issues/6596#issuecomment-1068965381). 
should we skip it for now? @individual-it @amrita-shrestha @saw-jan @SwikritiT or any suggestion?

**Comment by SwikritiT** [comment]
> According to fschade, e2e intermittent test will be solved by #6603 as mentioned in ([#6596 (comment)](https://github.com/owncloud/web/issues/6596#issuecomment-1068965381)). should we skip it for now? @individual-it @saw-jan @SwikritiT @kiranparajuli589 or any suggestion?

That PR has been merged so now you can rebase and push

File: .drone.star (hunk 4 above)
Comment (amrita-shrestha) [reply .drone.star]: I used the 2021 version in all cases and it working fine.

**Review by kiranparajuli589 — COMMENTED** [review]
lgtm :+1:

**Review by phil-davis — APPROVED** [review]
LGTM

File: .drone.star (hunk 3 above)
Comment (pascalwengerter) [reply .drone.star:870]: Okay, thanks for clarifying!:)

**Pull request merged**